content/post/against-ab-tests.md

```diff
@@ -0,0 +1,93 @@
+
+Yet Another Mattress Company (YAMC) sells mattresses online (you may have seen their advertisements on the subway). They want to test a new version of their checkout flow that is optimized for mobile. The designers are a little bit concerned that while this new version is less clunky, it also conveys less information during the checkout flow, so it could hurt conversion on desktop.
+
+The team launches the test, and here are the results:
+
+| Test Arm         | Impressions | Conversions | Conversion Rate |
+|------------------|-------------|-------------|-----------------|
+| Mobile Optimized | 10000       | 300         | 3.0%            |
+| Old Version      | 10000       | 300         | 3.0%            |
+
+Rats! No difference. On a hunch, you decide to split the results by mobile and desktop traffic.
+
+| Device Type | Test Arm         | Impressions | Conversions | Conversion Rate |
+|-------------|------------------|-------------|-------------|-----------------|
+| Mobile      | Mobile Optimized | 3000        | 250         | 8.3%            |
+| Mobile      | Old Version      | 3000        | 100         | 3.3%            |
+| Desktop     | Mobile Optimized | 2000        | 50          | 2.5%            |
+| Desktop     | Old Version      | 2000        | 200         | 10%             |
+
+Wow! The new version ... did exactly what the designers expected! It made things better for mobile users and worse for desktop users[^1].
+
+It is really too bad that our A/B test showed no effect. I guess we'll have to send our designers back to the drawing board...
+
+But wait! What if we serve the mobile-optimized version to customers browsing on mobile? And serve the desktop-optimized version to customers browsing on desktop? What if we found a landing page that worked better on weekends when people have more time read? What if we find advertising copy that works well in California but not in Massachusetts? 
+
+**What if websites didn't have to be one-size-fits all?**
+
+## Challenges
+
+I cannot tell whether this idea is obvious or revolutionary. It feels so obvious as to be almost silly, but if you look at the way most organizations build, test, and deploy software it is actually a pretty fundamental shift in the way we approach software problems.  
+In many organizations, there is still one version of the production web application. Tests may be run, but once a test is declared a winner or a loser, the losing versions are discarded and there is a new "king of the hill" one-true-production-application.
+
+In order to fully embrace the variability we actually see in our customers and our users, we have to build software in a fundamentally different way. We'll need new and better tools, and we will have to educate our stakeholders on this new way of thinking about the world.
```

Comment:
Unsure here, but is there any element of discrimination in content or ethical treatment re: dynamic pricing that you want to note? Or do we not care about that so long as we get the clicks?

Reply:
Willing to elide for now, but really important point!